veniq/dataset_collection/types_identifier.py

```diff
@@ -78,86 +82,233 @@ class InlineWithoutReturnWithoutArguments(IBaseInlineAlgorithm):
+        spaces_in_body = self.get_suitable_spaces(body_start_line, invocation_line, lines)
+        body_lines = [spaces_in_body + i for i in body_lines_without_spaces]
+        return body_lines
+
+    def get_lines_after_invocation(
+            self,
+            filename_out: pathlib.Path,
+            filename_in: pathlib.Path,
+            invocation_line: str
+    ) -> List[str]:
+        """
+        This function is aimed to obtain lines from the original
+        file after invocation line, which was detected.
+        Especially, it will be inserted after body of inlined method.
+        """
+        original_file = open(filename_in, encoding='utf-8')
+        lines = list(original_file)
+        lines_after_invoсation = lines[invocation_line:]
+        return lines_after_invoсation
+
+    def inline_function(
+            self,
+            filename_in: pathlib.Path,
+            invocation_line: int,
+            body_start_line: int,
+            body_end_line: int,
+            filename_out: pathlib.Path
+    ) -> None:
+        lines_of_final_file = []
         # original code before method invocation, which will be substituted
-        lines_before_invoсation = lines[:invocation_line - 1]
-        for i in lines_before_invoсation:
-            f_out.write(i)
+        lines_before_invoсation = self.get_lines_before_invocation(
+            filename_out,
+            filename_in,
+            invocation_line
+        )
+        lines_of_final_file += lines_before_invoсation
 
         # body of the original method, which will be inserted
-        num_spaces_before = len(lines[invocation_line - 1]) - len(lines[invocation_line - 1].lstrip(' '))
-        num_spaces_body = len(lines[body_start_line - 1]) - len(lines[body_start_line - 1].lstrip(' '))
-        body_lines = lines[body_start_line - 1:body_end_line]
-        for i in body_lines:
-            f_out.write(' ' * (num_spaces_before - num_spaces_body))
-            f_out.write(i)
+        body_lines = self.get_lines_of_method_body(
+            filename_out,
+            filename_in,
+            invocation_line,
+            body_start_line,
+            body_end_line
+        )
+        lines_of_final_file += body_lines
 
         # original code after method invocation
-        original_code_lines = lines[invocation_line:]
-        for i in original_code_lines:
-            f_out.write(i)
+        original_code_lines = self.get_lines_after_invocation(
+            filename_out,
+            filename_in,
+            invocation_line
+        )
+        lines_of_final_file += original_code_lines
 
+        f_out = open(filename_out, 'w', encoding='utf-8')
+        for line in lines_of_final_file:
+            f_out.write(line)
         f_out.close()
-        original_file.close()
 
 
 class InlineWithReturnWithoutArguments(IBaseInlineAlgorithm):
 
     def __init__(self):
         super().__init__()
 
-    def inline_function(
+    def get_lines_before_invocation(
```

Comment:
You have the same function. I see, we have a code duplication, extract this default functionality to a protected method in the base abstract class

Reply:
fixed